### PR TITLE
Don't invoke Maven every time we run a test

### PR DESCRIPTION
**Pull request by Baltoli**
Previously, the testing script would invoke Maven every time it compiled a pattern-matching tree; this was slow and made the CI process prone to intermittent network failures. This PR addresses the issue by instead directly invoking the compiled JAR file that we already built with Maven.

Fixes https://github.com/runtimeverification/llvm-backend/issues/1055

### CODE REVIEW

File: bin/llvm-kompile-testing [hunk 1]
```diff
@@ -11,7 +11,17 @@ trap 'rm -rf "$dt_dir"' INT TERM EXIT
...
 shift; shift
-( cd "@PROJECT_SOURCE_DIR@"/matching && mvn exec:java -Dexec.args="$definition qbaL $dt_dir 1" -q )
+
+matching_dir="@PROJECT_SOURCE_DIR@/matching"
+installed_jar="$matching_dir/target/llvm-backend-matching-1.0-SNAPSHOT-jar-with-dependencies.jar"
+
+if [ ! -f "$installed_jar" ]; then
+  echo "Pattern matching compiler isn't installed!"
+  echo "  looking for a jar file at $installed_jar"
+  exit 1
```
Comment (Robertorosmaninho) [inline bin/llvm-kompile-testing:21]: Why are we exiting here?

Comment (Baltoli) [reply bin/llvm-kompile-testing:21]: Because the pattern matching compiler jar isn't installed; how would you propose we continue execution?

Comment (Robertorosmaninho) [reply bin/llvm-kompile-testing:21]: The message above makes me feel that the script will look for the jar and not finish its execution...
I think I would create an else branch to build it as we were before, no?

Comment (Robertorosmaninho) [reply bin/llvm-kompile-testing:21]: Alright! Makes sense!

**Review by dwightguth — APPROVED** [review]

**Review by gtrepta — APPROVED** [review]

**Pull request merged**